concordium-consensus/src/Concordium/GlobalState.hs

```diff
@@ -401,7 +397,22 @@ class GlobalStateConfig (c :: ProtocolVersion -> Type) where
     -- |Generate context and state from the initial configuration. This may
     -- have 'IO' side effects to set up any necessary storage.
     -- This may throw a 'GlobalStateInitException'.
-    initialiseGlobalState :: IsProtocolVersion pv => c pv -> LogIO (GSContext c pv, GSState c pv, GSLogContext c pv)
+    --
+    -- The return value is 'Left' in case an existing state was found and
+    -- successfully loaded. Otherwise it is @Right k@ and the continuation @k@,
+    -- when given genesis data will create a new state instance in the provided
+    -- configuration. The reason for the added complexity, instead of just
+    -- requiring genesis data up-front, is that loading genesis data can be very
+    -- expensive and is not needed if the state already exists.
+    initialiseGlobalState :: IsProtocolVersion pv => c pv -> LogIO (Either (GSContext c pv, GSState c pv, GSLogContext c pv) (GenesisData pv -> LogIO (GSContext c pv, GSState c pv, GSLogContext c pv)))
+
+    -- |A wrapper around 'initialiseGlobalState' in case genesis data is already available.
+    initialiseGlobalStateWithGenesis :: IsProtocolVersion pv => GenesisData pv -> c pv -> LogIO (GSContext c pv, GSState c pv, GSLogContext c pv)
+    initialiseGlobalStateWithGenesis gd cfg = initialiseGlobalState cfg >>= \case
+        Left x -> return x
+        Right k -> k gd
+
+    activateGlobalState :: IsProtocolVersion pv => Proxy (c pv) -> GSContext c pv -> GSState c pv -> LogIO (GSState c pv)
```

Comment:
Please explain how this function is different from `initialiseGlobalState`. These two names appear very close semantically. When do I use one and when do I use another?

I see that there's a comment like the one I'm asking about here in `SkovConfiguration`, which is describing the use of `activateSkovState`.